src/DO/Sara/Core/ArrayIterators/ArrayIterators.hpp

```diff
@@ -86,10 +86,10 @@ namespace DO { namespace Sara {
                         vector_type& pos,
                         const vector_type& strides,
                         const vector_type& sizes)
-      : cur_ptr_{ ptr }
-      , cur_pos_{ pos }
-      , strides_{ strides }
-      , sizes_{ sizes }
+      : cur_ptr_(ptr)
+      , cur_pos_(pos)
+      , strides_(strides)
+      , sizes_(sizes)
```

Comment:
Very big lesson learnt here. Don't use initializer list when initializing reference to object.

According to:http://stackoverflow.com/questions/28717682/initialization-of-const-reference-member-in-initializer-list, because `sizes_` is a reference, then
- the copy constructor of `vector_type` will be invoked,
- then the copy constructor returns a **temporary** object initialized from `sizes`,
- then `sizes_` is a const reference bound to that temporary object,
- and this is wrong because, `sizes` will be a dangling reference when the constructor of the ArrayIterator object finishes: the temporary object gets deleted once the constructor finishes.